docs/introduction/index.md

```diff
@@ -0,0 +1,135 @@
+# First steps
+
+## Which API should I use?
+
+In shellfish, you have two different ways of doing things:
+
+- First, if you prefer a more concise, fluent syntax, use the extension methods (e.g. `path.read()`) by importing the `shellfish.syntax.all.*` package.
+
+- If you prefer calling static methods, use direct method calls on the `FilesOs` object (e.g., `FilesOs.read(path)`). You can also import all the functions inside the `shellfish.FilesOs` package if you don't want to call the `FilesOs` object every time (e.g., `read(path)`). 
+
+    For this documentation we'll call the methods on the `FilesOs` objects in the static variant to difference them from the extension methods.
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala 3
+import shellfish.syntax.all.*
+
+val path = Path("data/test.txt")
+
+for
+  file <- path.read
+  _    <- path.append("I'll place this here.")
+yield ()
+```
+
+@:choice(static)
+
+```scala 3
+import shellfish.FilesOs
+
+val path = Path("data/test.txt")
+
+for
+  file <- FilesOs.read(path)
+  _    <- FilesOs.append(path, "I'll place this here.")
+yield ()
+```
+
+@:@
+
+Which one you should use really depends on your preferences and choices; if you like the method style of calling functions directly to the object, stick to the extension methods syntax, if you rather prefer a more Haskell-like style, stick to the static object calls!
+
+## Imports
+
+If you just want to start scripting right away, importing `shellfish.*` will do the trick; it imports all the extension methods and functionality you need, such as types and functions, to start working right away.
+
+But if you want more concise functionality, the `shellfish.syntax.path.*` and `shellfish.syntax.process.*` will only import the extension methods.
+
+For the static methods, the `shellfish.FilesOs` and `shellfish.ProcessesOs` imports will provide the functions to work with files and processes, if that is your preferred style.
+
+Finally, `shellfish.domain.*` will import all the types needed to work with the library!
```

Comment:
I didn't see all this imports coming :P
In any case I'll say that we can completely skip this section, and let the users do everything with a single `shellfish.*` import